File: 1.2. Haskell/Task-1.2.hs
module Task21 where

isPow2 :: Int -> Bool
isPow2 1 = True
isPow2 n = n > 0 && even n && isPow2 (n `div` 2)

splitPow2 :: [Int] -> ([Int], [Int])
splitPow2 xs = (filter isPow2 xs, filter (not . isPow2) xs)

File: 1.2. Prolog/Task-1.2.pl
is_pow2(1).
is_pow2(N) :- N > 1, N mod 2 =:= 0, N1 is N // 2, is_pow2(N1).

split_pow2([], [], []).
split_pow2([H|T], [H|Pow], NotPow) :-
    is_pow2(H), !,
    split_pow2(T, Pow, NotPow).
split_pow2([H|T], Pow, [H|NotPow]) :-
    split_pow2(T, Pow, NotPow).
